client/client.go

```diff
@@ -48,6 +72,15 @@ func decodeWithNumber(r io.Reader, value interface{}) error {
 
 func unixDialer(socketPath string) func(string, string) (net.Conn, error) {
 	return func(_, _ string) (net.Conn, error) {
+		if _, err := os.Stat(socketPath); os.IsNotExist(err) {
+			return nil, &SocketNotFoundError{
+				Err:        err,
+				SocketPath: socketPath,
+			}
+		} else if err != nil {
+			return nil, fmt.Errorf("checking if pebble socket path %q exists: %w", socketPath, err)
+		}
```

Comment:
IMO this would be slightly clearer flattened as:

```
_, err := os.Stat(socketPath)
if errors.Is(err, fs.ErrNotExist) {
	return nil, &SocketNotFoundError{Err: err, SocketPath: socketPath}
}
if err != nil {
	return ...
}
```

Note also the use of `errors.Is`, which is what the Go docs now recommend for new code, instead of `os.IsNotExist`.